crates/voicevox_core_java_api/src/synthesizer.rs

```diff
@@ -0,0 +1,398 @@
+        if let Ok(cpu_num_threads) = cpu_num_threads.i() {
+            options.cpu_num_threads = cpu_num_threads as u16;
+        }
+
+        let load_all_models = env.get_field(&builder, "loadAllModels", "Z")?;
+        if let Ok(load_all_models) = load_all_models.z() {
+            options.load_all_models = load_all_models;
+        }
+
+        let open_jtalk = unsafe {
+            env.get_rust_field::<_, _, Arc<voicevox_core::OpenJtalk>>(&open_jtalk, "internal")?
+                .clone()
+        };
+        let internal = RUNTIME.block_on(voicevox_core::Synthesizer::new_with_initialize(
+            open_jtalk,
+            Box::leak(Box::new(options)),
+        ))?;
+        unsafe { env.set_rust_field(&this, "internal", Arc::new(Mutex::new(internal)))? };
+        Ok(())
+    })
+}
+
+#[no_mangle]
+pub extern "system" fn Java_jp_Hiroshiba_VoicevoxCore_Synthesizer_rsLoadVoiceModel<'local>(
+    env: JNIEnv<'local>,
+    this: JObject<'local>,
+    model: JObject<'local>,
+) {
+    throw_if_err(env, (), |env| {
+        let model = unsafe {
+            env.get_rust_field::<_, _, Arc<voicevox_core::VoiceModel>>(&model, "internal")?
+                .clone()
+        };
+        let internal = unsafe {
+            env.get_rust_field::<_, _, Arc<Mutex<voicevox_core::Synthesizer>>>(&this, "internal")?
+                .clone()
+        };
+        {
+            let mut internal = internal.lock().unwrap();
+            RUNTIME.block_on(internal.load_voice_model(&model))?;
+        }
+        Ok(())
+    })
+}
+
+#[no_mangle]
+pub extern "system" fn Java_jp_Hiroshiba_VoicevoxCore_Synthesizer_rsUnloadVoiceModel<'local>(
+    env: JNIEnv<'local>,
+    this: JObject<'local>,
+    model_id: JString<'local>,
+) {
+    throw_if_err(env, (), |env| {
+        let model_id: String = env.get_string(&model_id)?.into();
+
+        let internal = unsafe {
+            env.get_rust_field::<_, _, Arc<Mutex<voicevox_core::Synthesizer>>>(&this, "internal")?
+                .clone()
+        };
+
+        {
+            let mut internal = internal.lock().unwrap();
+
+            internal.unload_voice_model(&voicevox_core::VoiceModelId::new(model_id))?;
+        }
+
+        Ok(())
+    })
+}
+
+#[no_mangle]
+pub extern "system" fn Java_jp_Hiroshiba_VoicevoxCore_Synthesizer_rsIsLoadedVoiceModel<'local>(
+    env: JNIEnv<'local>,
+    this: JObject<'local>,
+    model_id: JString<'local>,
+) -> jboolean {
+    throw_if_err(env, false, |env| {
+        let model_id: String = env.get_string(&model_id)?.into();
+
+        let internal = unsafe {
+            env.get_rust_field::<_, _, Arc<Mutex<voicevox_core::Synthesizer>>>(&this, "internal")?
+                .clone()
+        };
+
+        let is_loaded = {
+            let internal = internal.lock().unwrap();
+            internal.is_loaded_voice_model(&voicevox_core::VoiceModelId::new(model_id))
+        };
+
+        Ok(is_loaded)
+    }) as jboolean
```

Comment:
(少なくとも`Into`でできる範囲については、) 数値型の変換を`as`でやるのはできる限り避けた方がいいかと思います。

```suggestion
    })
    .into()
```